services/src/main/java/org/keycloak/broker/oidc/mappers/ClaimToUserSessionNoteMapper.java

```diff
@@ -0,0 +1,123 @@
+package org.keycloak.broker.oidc.mappers;
+
+import org.keycloak.broker.oidc.KeycloakOIDCIdentityProviderFactory;
+import org.keycloak.broker.oidc.OIDCIdentityProviderFactory;
+import org.keycloak.broker.provider.BrokeredIdentityContext;
+import org.keycloak.models.Constants;
+import org.keycloak.models.IdentityProviderMapperModel;
+import org.keycloak.models.IdentityProviderSyncMode;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.RealmModel;
+import org.keycloak.models.UserModel;
+import org.keycloak.provider.ProviderConfigProperty;
+import org.keycloak.util.JsonSerialization;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class ClaimToUserSessionNoteMapper extends AbstractClaimMapper {
+
+    private static final String CLAIM_PROPERTY_NAME = "claims";
+    private static final String[] COMPATIBLE_PROVIDERS = {KeycloakOIDCIdentityProviderFactory.PROVIDER_ID,
+            OIDCIdentityProviderFactory.PROVIDER_ID};
+
+    private static final List<ProviderConfigProperty> CONFIG_PROPERTIES = new ArrayList<>();
+
+    private static final Set<IdentityProviderSyncMode> IDENTITY_PROVIDER_SYNC_MODES =
+            new HashSet<>(Arrays.asList(IdentityProviderSyncMode.values()));
+
+    static {
+        ProviderConfigProperty claimsProperty = new ProviderConfigProperty();
+        claimsProperty.setName(CLAIM_PROPERTY_NAME);
+        claimsProperty.setLabel("Claims");
+        claimsProperty.setHelpText(
+                "Name of the claims to search for in the token. You can reference nested claims using a '.', i.e. 'address.locality'. To use dot (.) literally, escape it with backslash (\\.)");
+        claimsProperty.setType(ProviderConfigProperty.MULTIVALUED_STRING_TYPE);
+        CONFIG_PROPERTIES.add(claimsProperty);
+    }
+
+    public static final String PROVIDER_ID = "oidc-user-session-note-idp-mapper";
+
+    @Override
+    public String[] getCompatibleProviders() {
+        return COMPATIBLE_PROVIDERS;
+    }
+
+    @Override
+    public String getDisplayCategory() {
+        return "User Session";
+    }
+
+    @Override
+    public String getDisplayType() {
+        return "User Session Note Mapper";
+    }
+
+    @Override
+    public String getHelpText() {
+        return "Add every matching claim to the user session note.";
```

Comment:
Is it possible to add possibly another sentence like: `This can be used together for instance with 'User Session Note' protocol mapper configured for your client scope or client, so that claims for 3rd party IDP would be available in the tokens sent to your client application.`

I mean this use-case is a bit advanced and will be nice to have some more "documentation hints" for people wondering about the typical usage of this capability.